pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -516,9 +516,10 @@ func (r *ReconcileBareMetalHost) actionInspecting(prov provisioner.Provisioner,
 	}
 
 	if provResult.ErrorMessage != "" {
-		info.host.Status.Provisioning.State = metal3v1alpha1.StateRegistrationError
```

Comment:
I know you mention it in the commit message, but I wonder why not keep this in place?

Reply:
It's the wrong state anyway (i.e. the error is not due to a registration problem), and our ultimate goal is to get rid of these error states altogether (in #285), so introducing a new place where we switch to this state (which this effectively would be, since the code did nothing before) makes things worse and our future plans more complicated.